### DISCUSSION TITLE
Trixter xdream software hangs on startup

### DISCUSSION THREAD

**Discussion by A-Zcomputers** [xdream-biking/trixter-xdream-bike#107] [Q&A]
Work in a small computer store, customer brought in a trixter xdream bike, with a dell aio attached to it. Windows was corrupt and required reinstallation. Was able to grab the trixter data from the windows.old folder, have installed all that i can find within the trixter/live folder. Serial to usb/prosgres/vlc/smartfox.
When i launch the trixter executable it will only load  the initial launch screen and proceed no further. 
I suspect this is due to having installed the software as such. Given there is nowhere anywhere that has this installer, would anyone be willing to supply it if i can show proof that i have the original files, the bike itself and the original dell aio that was shipped with the bikes? 
Im going in circles and the customer is desperate to have it up and running again.

**Comment by drmason789** [comment]
Hi, this discussion forum is for the Trixter X-Dream v1 bike, which has a Dell Optiplex PC in a curved metal box out in front of the bike, and a 17" monitor, and v1 to v3.* of the original X-Dream mountain biking software.

There was a later version of the whole system (v2) which is more compact, with an all-in-one PC at the front, and updated software that apparently incorporates recorded video as well as computer generated. I personally have never seen one of these newer ones in person, and know very little of them currently. I have seen a couple on eBay in the UK, but most trades I observe there are for the v1 system. Considering that the Dell Aio is an all-in-one, I think you're dealing with a v2 system.

I don't have an installer for either of them. Everything regarding the software in these discussions I have worked out from an installation on a v1 system.

However, it doesn't mean I can't help you get it working.

**Comment by drmason789** [comment]
Could you define "hangs on startup" more precisely? 
e.g. does the computer freeze and require a reset?
Can you alt-F4 out of the application?
Does it appear active then freeze doing something that looks like it should continue, e.g. showing an animation that freezes, or playing music that stops abruptly?

I have [here](https://youtu.be/0BZGE5vtdoI) a video showing what it's like to log into the software from the v1 system. There is a pause on the start screen, then a yellow button appears, saying press the green button to continue. The software would seem to freeze at this point if the bike's button press signal doesn't get to the software, e.g. because the OS has the wrong COM port configured.

**Reply by A-Zcomputers** [reply]
Thankyou for the swift response David,
It most certainly must be a V2 
Upon launching the trixter application it presents a screen with the trixter logo greyed out and a message saying something along the lines of please wait a moment and never progresses. If i was to take a guess the trixter logo would be a loading bar generally.
The machine is still responsive, can alt tab, alt F4 etc.
I have launched the diagnostic utility and the bike is responsive on COM3. 
I feel as though its missing some required initial setup files, rather then just simply copying over the trixter folder. To give perspective the folder size was roughly 53gb, pretty substantial!

**Reply by drmason789** [reply]
Well, at least the bike is responsive. The software for the v1 system defaults to com2, but this can be configured, see #106, perhaps there is something similar for the newer software, though it's probably better to just change the COM port the USB to serial port is emulating, see #6 - changing the COM port should be similar.

The old software survives not having a bike connected though, so I think you probably have a different problem. I would install the Sysinternals Suite (free Microsoft tools) and run procmon, filtering out all items that are not from the application and see if it's trying to do something and failing, such as loading a DLL that isn't installed, e.g. a media codec or VC runtime DLL.

**Reply by drmason789** [reply]
If it's trying to access postgresql, it could be that postgresql's security is blocking access. In my instructions for Windows I say to copy a config file but in my Linux instructions #105 I'm more specific about what to edit.

**Reply by drmason789** [reply]
Also, be sure to get the "hosts" file from the windows.old folder, it's usually under "C:\Windows\System32\drivers\etc"

In my v1 system this file looks like this:

```
192.168.32.254	      	router		
192.168.32.1          	trixter01
192.168.32.2          	trixter02
192.168.32.3          	trixter03
192.168.32.4          	trixter04
192.168.32.5          	trixter05
192.168.32.6          	trixter06
192.168.32.7          	trixter07
192.168.32.8          	trixter08
192.168.32.9          	trixter09
192.168.32.10         	trixter10
192.168.32.11         	trixter11
192.168.32.12         	trixter12
192.168.32.13         	trixter13
192.168.32.14         	trixter14
192.168.32.15		trixter15
```
The application may be trying to use these network names.

**Comment by drmason789** [comment]
For the PostgreSQL config (_installdir_/data/pg_hba.conf, the default has a pair of lines that look like this:

```
# TYPE  DATABASE        USER            ADDRESS                 METHOD
# IPv4 local connections:
host    all             all             127.0.0.1/32            md5
# IPv6 local connections:
host    all             all             ::1/128                 md5
```

But in the one provided with the v1 system, these lines are like this:


```
# TYPE  DATABASE        USER            ADDRESS                 METHOD
# IPv4 local connections:
host    all             all             127.0.0.1/32            trust
# IPv6 local connections:
host    all             all             ::1/128                 trust
```

I.e. "md5" changed to "trust"
